### PR TITLE
Update README to include discussion of how changes impact deployment

### PR DESCRIPTION
**Pull request by woodt**
Thought it might be worthwhile to capture these guidelines somewhere.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,32 @@ The _only_  deployable artifact associated with this repository is the
...
+
+- If you need to add or change a dependency, you should make that
+  change in the `requirements/Pipfile`, run `make update-dependencies`
+  and commit the changed files.  (See the section below on
+  requirements for details.)  Good news: no other changes are required!
+  
+- If you need to add or remove a plugin, you will also need to update
+  the plugin list in
+  [datagov-deploy](https://github.com/GSA/datagov-deploy) Currently,
+  that means updating the value of `catalog_ckan_plugsin_default` in
```
Comment (starsinmypockets) [inline README.md:28]:
```suggestion
  that means updating the value of `catalog_ckan_plugins_default` in
```

File: README.md [hunk 2]
```diff
@@ -13,6 +13,32 @@ The _only_  deployable artifact associated with this repository is the
 `requirements-freeze.txt` file. See [datagov-deploy](https://github.com/GSA/datagov-deploy)
 for full configuration in live environments.
 
+The live environment is different than the development environment in a number of ways,
+so it is important to recognize how changes made to the application need to be captured
```
Comment (starsinmypockets) [inline README.md:17]:
```suggestion
so it is important to capture these changes.
```

File: README.md [hunk 3]
```diff
@@ -13,6 +13,32 @@ The _only_  deployable artifact associated with this repository is the
...
+so that the application is deployed properly.
```
Comment (starsinmypockets) [inline README.md:18]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -13,6 +13,32 @@ The _only_  deployable artifact associated with this repository is the
...
+  
+- If you need to add or change configuration that lives in the
+  application *ini* file, you will also need to update the
+  configuration file template in
+  [datagov-deploy](https://github.com/GSA/datagov-deploy) Currently,
+  this means modifying
+  `ansible/roles/software/ckan/catalog/ckan-app/templates/catalog-next/etc_ckan_production_ini.j2`.
+  
+- If you find you need to modify the `ckan/Dockerfile` to add OS
+  packages or install software, other changes may be needed to be made
```
Comment (starsinmypockets) [inline README.md:39]:
```suggestion
  packages or install software, other changes may need to be made
```

File: README.md (hunk 1 above)
Comment (woodt) [reply README.md:28]: Fixed.

File: README.md (hunk 2 above)
Comment (woodt) [reply README.md:17]: Went with slightly different language :-)

File: README.md (hunk 4 above)
Comment (woodt) [reply README.md:39]: Fixed.

File: README.md (hunk 3 above)
Comment (woodt) [reply README.md]: I kind of like leaving in the idea that the additional work is focused on the goal that the application is deployable.

**Pull request merged**